Softeq.XToolkit.WhiteLabel.Droid/Navigation/PageNavigationService.cs

```diff
@@ -13,23 +15,44 @@ namespace Softeq.XToolkit.WhiteLabel.Droid.Navigation
 {
     public class PageNavigationService : IPageNavigationService
     {
-        private readonly Stack<string> _backStack;
+        private readonly Stack<IViewModelBase> _backStack;
         private readonly ViewLocator _viewLocator;
+        private readonly IJsonSerializer _jsonSerializer;
 
-        public PageNavigationService(ViewLocator viewLocator)
+        public PageNavigationService(ViewLocator viewLocator, IJsonSerializer jsonSerializer)
         {
             _viewLocator = viewLocator;
-            _backStack = new Stack<string>();
+            _jsonSerializer = jsonSerializer;
+            _backStack = new Stack<IViewModelBase>();
         }
 
-        public int BackStackCount => _backStack.Count;
+        public bool CanGoBack => !CrossCurrentActivity.Current.Activity.IsTaskRoot;
```

Comment:
every activity can set `ActivityFlags.NewTask` flag, it will be root after that?

Reply:
https://developer.android.com/guide/components/activities/tasks-and-back-stack
Yauhen Sampir, [Nov 26, 2018 at 4:57:02 PM]:
Using Intent flags
When you call startActivity(), you can include a flag in the Intent that declares how (or whether) the new activity should associate with the current task.

Using the manifest file
When you declare an activity in your manifest file, you can specify how the activity should associate with tasks when it starts.